iot/api/CloudIotSample/Program.cs

```
@@ -1209,6 +1236,142 @@ public static object UnbindDeviceFromGateway(string projectId, string cloudRegio
+            var fullpath = $"projects/{projectId}/locations/{cloudRegion}/registries/{registryId}/devices/{deviceId}";
+            try
+            {
+                var device = cloudIot.Projects.Locations.Registries.Devices.Get(fullpath).Execute();
+                Console.WriteLine("Devices -->: {0}", parent);
+
+                if (device != null)
+                {
+                    bool isGateway = device.GatewayConfig.GatewayType == "GATEWAY";
+                    if (!isGateway)
+                    {
+                        //get list of gateways this non-gateway device bound to.
+                        //cloudIot.Projects.Locations.Registries.Devices
+                        var req = cloudIot.Projects.Locations.Registries.Devices.List(parent);
+                        req.GatewayListOptionsAssociationsDeviceId = device.Id;
+                        var gateways = req.Execute().Devices;
+                        if (gateways != null)
+                        {
+                            gateways.ToList().ForEach(gateway =>
+                                {
+                                    UnbindDeviceFromGatewayRequest unbindReq = new UnbindDeviceFromGatewayRequest
+                                    {
+                                        DeviceId = device.Id,
+                                        GatewayId = gateway.Id
+                                    };
+                                    try
+                                    {
+                                        cloudIot.Projects.Locations.Registries.UnbindDeviceFromGateway(unbindReq, parent).Execute();
+                                        Console.WriteLine("Unbound device from the gateway {0}", gateway.Id);
+                                    }
+                                    catch (Google.GoogleApiException e)
+                                    {
+                                        Console.WriteLine("Could not unbind device");
```

The code will continue unbinding devices even if one unbind operation fails.  That looks good to me.  But we hide failures from the caller.  We can still throw an AggregateException afterwards if something failed.

---

I will add AggregateException afterwards.